src/Microsoft.ML.AutoML/Utils/UserInputValidationUtil.cs

```diff
@@ -57,6 +57,14 @@ public static void ValidateNumberOfCVFoldsArg(uint numberOfCVFolds)
             }
         }
 
+        public static void ValidateSamplingKey(string samplingKeyColumnName, string groupIdColumnName, TaskKind task)
+        {
+            if (task == TaskKind.Ranking && samplingKeyColumnName != null && samplingKeyColumnName != groupIdColumnName)
+            {
+                throw new ArgumentException($"{nameof(samplingKeyColumnName)} must be the same as {nameof(groupIdColumnName)} for Ranking Experiments", samplingKeyColumnName);
```

Comment:
Nit: "If provided, samplingKeyColumnName must be the same as..."
I think that the current wording is a little bit misleading, since on this PR we're allowing users to set the GroupId column without providing the samplingKeyColumn.